### PR TITLE
Update snapcraft.yaml

### PR DESCRIPTION
**Pull request by simos**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

I updated the snapcraft.yaml configuration file so that the snapcraft tool can create a snap package for radare2.

* Introduces the "devmode" confinement. This allows the snap package to work without confinement. 
* Switched base to "core18". More recent base, matching the Ubuntu 18.04 runtime. The default is the Ubuntu 16.04 runtime.
* Installs ninja and acr as build dependencies. The versions selected in the configuration work for me. You may want to bump them to more recent versions. 

...

**Test plan**

I have tested this configuration on Ubuntu 18.04. I installed `snapcraft` and run `snapcraft` while in the `radare2` source directory. 
After a while, snapcraft created the snap package in the current directory. 
I then, 

```
sudo snap install radare2_4.5.0_amd64.snap --devmode
```
Finally, I tested that radare2 works by attaching to a PID of another process (so that confinement is not blocking radare2).
```
sudo radare2.radare2 -d 1234
```

**Closing issues**

Reference: https://github.com/radareorg/radare2/issues/10428

### CODE REVIEW

File: snapcraft.yaml [hunk 1]
```diff
@@ -29,9 +37,6 @@ apps:
   rasm2:
     command: usr/bin/rasm2
     plugs: [home]
-  sdb:
-    command: usr/bin/sdb
-    plugs: [home]
   rabin2:
```
Comment (XVilka) [inline snapcraft.yaml:40]: I think it also makes sense to add `rasign2` command, since it was reintroduced recently.

File: snapcraft.yaml [hunk 2]
```diff
@@ -60,3 +65,35 @@ parts:
...
       - python3-pip
       - python3-setuptools
+      - libmagic-dev
+      - pkg-config
+      - libzip-dev
+      - liblz4-dev
+      - libssl-dev
+      - libbsd-dev
+      - libcapstone-dev
+      - cmake
```
Comment (ret2libc) [inline snapcraft.yaml:75]: why cmake?

File: snapcraft.yaml [hunk 3]
```diff
@@ -60,3 +65,35 @@ parts:
...
+      - libuv1-dev
+    stage-packages:
+      - libuv1
+    after: [ninja, acr]
+  ninja:
```
Comment (ret2libc) [inline snapcraft.yaml:80]: I don't know anything about snap... but why are you installing both ninja and acr?

File: snapcraft.yaml (hunk 1 above)
Comment (simos) [reply snapcraft.yaml:42]: I added `rasign2` in the new commit.

File: snapcraft.yaml (hunk 2 above)
Comment (simos) [reply snapcraft.yaml]: `cmake` was a remnant from an earlier version of the snapcraft.yaml configuration. 
I removed it. 

The `build-packages` are packages that are installed to aid in the compilation of the software.

File: snapcraft.yaml (hunk 3 above)
Comment (simos) [reply snapcraft.yaml]: Both are remnants from an older version of the `snapcraft.yaml`, circa two years ago.
I removed them.

**Comment by simos** [comment]
I have updated the `snapcraft.yaml` file. 

For my tests I am using this repository, https://github.com/simos/radare2-simosx
and I have linked it to the Snapcraft Build server for my test package `radare2-simosx`.

```
$ sudo snap install radare2-simosx
error: snap "radare2-simosx" is not available on stable but is available to
       install on the following channels:

       candidate  snap install --candidate radare2-simosx
       beta       snap install --beta radare2-simosx
       edge       snap install --edge radare2-simosx

       Please be mindful pre-release channels may include features not
       completely tested or implemented. Get more information with 'snap info
       radare2-simosx'.

$ snap install radare2-simosx --channel=beta
error: The publisher of snap "radare2-simosx" has indicated that they do not
       consider this revision to be of production quality and that it is only
       meant for development or testing at this point. As a consequence this
       snap will not refresh automatically and may perform arbitrary system
       changes outside of the security sandbox snaps are generally confined to,
       which may put your system at risk.

       If you understand and want to proceed repeat the command including
       --devmode; if instead you want to install the snap forcing it into
       strict confinement repeat the command including --jailmode.

$ snap install radare2-simosx --channel=beta --devmode
radare2-simosx (beta) 4.5.0 from Simos Xenitellis (simosx) installed
$ radare2-simosx.radare2 -v
radare2 4.5.0 24990 @ linux-x86-64 git.4.5.0
commit: 9d7eda5ec7367d1682e489e92d1be8e37e459296 build: 2020-07-23__10:59:04
$ sudo radare2-simosx.radare2 -d 1786
= attach 1786 1786
bin.baddr 0x56163ccab000
Using 0x56163ccab000
asm.bits 64
[0x7fc153b29d27]> q
Do you want to quit? (Y/n) 
Do you want to kill the process? (Y/n) n
$ 
```

File: snapcraft.yaml [hunk 4]
```diff
@@ -47,11 +54,14 @@ apps:
...
     plugs: [home]
+  rasign2:
+    command: usr/bin/rasign2
+    plugs: [home]
 
 parts:
   radare2:
-    source: https://github.com/radareorg/radare2.git
-    source-tag: '2.6.0'
+    source: https://github.com/radare/radare2.git
```
Comment (XVilka) [inline snapcraft.yaml:63]: No, it's actually `https://github.com/radareorg/radare2.git`

Comment (simos) [reply snapcraft.yaml]: I changed the URL and added the documentation.

**Pull request merged**